### PR TITLE
feat: add continuous deployment workflow

### PR DESCRIPTION
**Pull request by gtirloni**
* [x] I've read the [guidelines for Contributing to the Platform Co-op Toolkit](https://github.com/platform-coop-toolkit/.github/blob/master/CONTRIBUTING.md)
* [x] This isn't a duplicate of an existing pull request

## Description

Add GitHub Actions workflows that will do continuous deployment of the {demo,dev} websites.

## Additional information

Before this works, the Secrets need to be added to the repository.

### CODE REVIEW

**Comment by gtirloni** [comment]
> @gtirloni This looks great. There's one additional step which I expect will be needed, namely running `python3 manage.py migrate` to handle any new database migrations on deploy

I've added the command to the workflow. Thanks for the heads up!

**Comment by gtirloni** [comment]
> I don't see a step here that recompiles static assets. I don't know if that's because this installation is using `runserver`, which is not recommended for any use other than local development and is not how the AWS server was setup. The AWS setup used `honcho` (a Python port of Foreman) with a `Procfile` and `gunicorn`. `honcho` can, I believe, generate starter configs to be used with `systemctl` for a more permanent & robust solution but the `honcho` approach is what would be required if this moves back to heroku.

All of that is contained in the Dockerfile. The application is running inside a Docker container and that is behind an Nginx proxy as well.

**Comment by erictheise** [comment]
What I see in your `Dockerfile` is [`runserver`](https://github.com/platform-coop-toolkit/coop-map-directory-index/blob/e234dff929f012835e0e795c8911dfc77c19f848/Dockerfile#L35) which [should not be used in production](https://docs.djangoproject.com/en/3.0/ref/django-admin/#runserver) and what I don't see in the `Dockerfile` is any mention of `collectstatic`.

I've always thought of the AWS server as production-ready and that's how I configured it. I cannot recommend provisioning an Internet-facing server using a pure Python development server that "has not gone through security audits or performance tests".

File: .github/workflows/stack_demo.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            cd /srv/demo.directory.platform.coop                          && \
+            git checkout master                                           && \
+            git pull                                                      && \
+            source deploy.env                                             && \
+            /usr/local/bin/docker-compose -p $PROJECT_ID                     \
+                                          -f docker-compose.yml              \
+                                          up --force-recreate --build -d  && \
+            /usr/local/bin/docker-compose -p $PROJECT_ID                     \
+                                          -f docker-compose.yml              \
+                                          run app python3 manage.py migrate
```
Comment (greatislander) [inline .github/workflows/stack_demo.yml:35]: Looking at the Dockerfile I don't think this needs to be `python3` (I see you ran `collectstatic` with `python`, @gtirloni).

File: .github/workflows/stack_dev.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            cd /srv/dev.directory.platform.coop                           && \
+            git checkout dev                                              && \
+            git pull                                                      && \
+            source deploy.env                                             && \
+            /usr/local/bin/docker-compose -p $PROJECT_ID                     \
+                                          -f docker-compose.yml              \
+                                          up --force-recreate --build -d  && \
+            /usr/local/bin/docker-compose -p $PROJECT_ID                     \
+                                          -f docker-compose.yml              \
+                                          run app python3 manage.py migrate
```
Comment (greatislander) [inline .github/workflows/stack_dev.yml:35]: Doesn't need to be `python3` (see above).

File: .github/workflows/stack_demo.yml (hunk 1 above)
Comment (gtirloni) [reply .github/workflows/stack_demo.yml]: Changed, thanks!

File: .github/workflows/stack_dev.yml (hunk 2 above)
Comment (gtirloni) [reply .github/workflows/stack_dev.yml]: Changed, thanks!

**Comment by gtirloni** [comment]
I'll merge this so we can start to use it and find any bugs. I'll address any concerns in a future PR.

**Pull request merged**